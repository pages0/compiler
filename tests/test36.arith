let x : <int> = ref 0 in
let z : <int> = ref 0 in
while (!x <10) do
      let y : <int> =ref 0 in
      while (!y <10) do
      y := !y +1
      end ;
      z:= !z+!y;
      x := !x +1
      end ; !z