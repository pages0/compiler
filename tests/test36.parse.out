( let [ x ( ref 0 ) ] ( let [ z ( ref 0 ) ] ( ; ( while ( < ( ! x ) 10 ) ( let [ y ( ref 0 ) ] ( ; ( ; ( while ( < ( ! y ) 10 ) ( != y ( + ( ! y ) 1 ) ) ) ( != z ( + ( ! z ) ( ! y ) ) ) ) ( != x ( + ( ! x ) 1 ) ) ) ) ) ( ! z ) ) ) )